code/game/objects/structures/mannequin.dm

```diff
@@ -0,0 +1,1386 @@
+		else
+			qdel(clothToWear)
+	if(mapping_hat)
+		var/obj/item/clothToWear = new mapping_hat(src)
+		if(canEquip(null, SLOT_MANNEQUIN_HEAD, clothToWear))
+			clothing[SLOT_MANNEQUIN_HEAD] = clothToWear
+		else
+			qdel(clothToWear)
+	if(mapping_mask)
+		var/obj/item/clothToWear = new mapping_mask(src)
+		if(canEquip(null, SLOT_MANNEQUIN_MASK, clothToWear))
+			clothing[SLOT_MANNEQUIN_MASK] = clothToWear
+		else
+			qdel(clothToWear)
+	if(mapping_back)
+		var/obj/item/clothToWear = new mapping_back(src)
+		if(canEquip(null, SLOT_MANNEQUIN_BACK, clothToWear))
+			clothing[SLOT_MANNEQUIN_BACK] = clothToWear
+		else
+			qdel(clothToWear)
+	if(mapping_id)
+		var/obj/item/clothToWear = new mapping_id(src)
+		if(canEquip(null, SLOT_MANNEQUIN_ID, clothToWear))
+			clothing[SLOT_MANNEQUIN_ID] = clothToWear
+		else
+			qdel(clothToWear)
+
+	if(mapping_hand_right)
+		var/obj/item/clothToWear = new mapping_hand_right(src)
+		held_items[1] = clothToWear
+	if(mapping_hand_left)
+		var/obj/item/clothToWear = new mapping_hand_left(src)
+		held_items[2] = clothToWear
+
+	update_icon()
+
+
+/obj/structure/mannequin/proc/spin()
+	change_dir(turn(dir, 90))
+
+/obj/structure/mannequin/verb/rotate_mannequin()
+	set name = "Rotate Mannequin"
+	set category = "Object"
+	set src in oview(1)
+
+	if(!usr || !isturf(usr.loc))
+		return
+
+	if(anchored)
+		return
+
+	if(usr.isUnconscious() || usr.restrained())
+		return
+
+	spin()
+
+
+/obj/structure/mannequin/fat
+	name = "fat human marble mannequin"
+	icon_state="mannequin_marble_fat_human"
+	fat = 1
+
+/obj/structure/mannequin/vox
+	name = "vox marble mannequin"
+	icon_state="mannequin_marble_vox"
+	species_type = /datum/species/vox
+
+/obj/structure/mannequin/wood
+	name = "human wooden mannequin"
+	desc = "This should look great in a visual arts workshop."
+	icon_state="mannequin_wooden_human"
+	health = 30
+	maxHealth = 30
+
+/obj/structure/mannequin/wood/Destroy()
+	getFromPool(/obj/item/stack/sheet/wood, loc, 5)//You get half the materials used to make a block back
+	..()
+
+/obj/structure/mannequin/wood/fat
+	name = "fat human wooden mannequin"
+	icon_state="mannequin_wooden_fat_human"
+	fat = 1
+
+/obj/structure/mannequin/wood/vox
+	name = "vox wooden mannequin"
+	icon_state="mannequin_wooden_vox"
+	species_type = /datum/species/vox
+
+
+
+
+
+/////////////////////////////////////////////////////////BLOCKS//////////////////////////////////////////////////////////
+//Use a chisel on those to sculpt them into mannequins.
+
+/obj/structure/block
+	name = "marble block"
+	desc = "Grab your chisel and get to work!"
+	anchored = 0
+	density = 1
+	icon = 'icons/obj/mannequin.dmi'
+	icon_state = "marble"
+	var/time_to_sculpt = 200
+	var/result = "/obj/structure/mannequin"
+	var/no_drop = 0
+
+/obj/structure/block/attackby(var/obj/item/weapon/W,var/mob/user)
+	if(iswrench(W))
+		return wrenchAnchor(user, 50)
+	else if(istype(W, /obj/item/weapon/chisel))
+
+		var/list/available_sculptures = list(
```

Comment:
I see.

By the way @RemieRichards I've meant to ask, do you regularly check every coderbase's github? For a tg coder I've seen you here pretty often.